core/src/consensus.rs

```diff
@@ -75,7 +75,8 @@ pub fn relative_min_window_density(b1: &MinaConsensusState, b2: &MinaConsensusSt
 
     let projected_window = {
         // Compute shift count
-        let shift_count = (max_slot - global_slot(b1) - 1).clamp(0, SUB_WINDOWS_PER_WINDOW);
+        let shift_count = 
+            (max_slot.saturating_sub(global_slot(b1) + 1)).clamp(0, SUB_WINDOWS_PER_WINDOW);
```

Comment:
Thinking about it a bit more, we don't even need `clamp` now that we use `saturating_sub` and it can be simplified a bit:

```suggestion
        let shift_count = max_slot
            .saturating_sub(global_slot(b1) + 1)
            .min(SUB_WINDOWS_PER_WINDOW);
```